Mathlib/Combinatorics/SimpleGraph/UniversalVerts.lean

```diff
@@ -51,4 +52,25 @@ lemma Subgraph.IsMatching.exists_of_universalVerts [Finite V] {s : Set V}
   have hadj (v : s) : G.Adj v (f v) := ht.1 (f v).2 (hd.ne_of_mem (f v).2 v.2)
   exact Subgraph.IsMatching.exists_of_disjoint_sets_of_equiv hd.symm f hadj
 
+lemma disjoint_image_val_universalVerts (s : Set G.deleteUniversalVerts.verts) :
+    Disjoint (Subtype.val '' s) G.universalVerts := by
+  simpa [deleteUniversalVerts, Subgraph.deleteVerts_verts, ← Set.disjoint_compl_right_iff_subset,
+    Set.compl_eq_univ_diff] using Subtype.coe_image_subset _ s
+
+/-- A component of the graph with universal vertices is even if we remove a set of representatives
+of odd components and a subset of universal vertices.
+
+This is because the number of vertices in the even components is not affected, and from odd
+components exactly one vertex is removed. -/
+lemma even_ncard_image_val_supp_sdiff_image_val_rep_union {t : Set V}
+    {s : Set G.deleteUniversalVerts.verts} (K : G.deleteUniversalVerts.coe.ConnectedComponent)
+    (h : t ⊆ G.universalVerts)
+    (hrep : ConnectedComponent.Represents s G.deleteUniversalVerts.coe.oddComponents) :
+    Even (Subtype.val '' K.supp \ (Subtype.val '' s ∪ t)).ncard := by
+  simp only [← Set.diff_inter_diff, ← Set.image_diff Subtype.val_injective,
+    sdiff_eq_left.mpr <| Set.disjoint_of_subset_right h (disjoint_image_val_universalVerts _),
+    Set.inter_diff_distrib_right, Set.inter_self, Set.diff_inter_self_eq_diff,
+    ← Set.image_inter Subtype.val_injective, Set.ncard_image_of_injective _ Subtype.val_injective,
+    K.even_ncard_supp_sdiff_rep hrep]
```

Comment:
Is there any chance you can unsqueeze that simp for robustness?

Reply:
Done:)